validphys2/src/validphys/process_options.py

```diff
@@ -256,6 +257,20 @@ def _dybosonpt_xq2map(kin_dict):
     return x, ET2
 
 
+def _dymll_xq2map(kin_info):
+    """
+    Computes x and q2 mapping for DY Z -> 2 leptons mass.
+    Here we calculate M_ll^2 and sqrt(s).
+    """
+
+    M_ll = kin_info.get_one_of(_Vars.M_ll)
+    sqrts = kin_info.get_one_of(_Vars.sqrts)
+    M_ll2 = M_ll**2
+    x = np.sqrt(M_ll) / sqrts
```

Comment:
I think this should be instead
```suggestion
    x = M_ll / sqrts
```

(mll^2 = x1*x2*s ->  x = mll/sqrts)

x = sqrt(x1*x2) is probably a bad approximation but it is the best we can get for this dataset with the information we have.